tests/Fixer/Basic/NoTrailingCommaInSinglelineFixerTest.php

```diff
@@ -0,0 +1,428 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\Basic;
+
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+
+/**
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\Basic\NoTrailingCommaInSinglelineFixer
+ */
+final class NoTrailingCommaInSinglelineFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @dataProvider provideFixCases
+     */
+    public function testFix(string $expected, ?string $input = null): void
+    {
+        $this->doTest($expected, $input);
+    }
+
+    public function provideFixCases(): iterable
+    {
+        yield [
+            '<?php [$x, $y] = $a;',
+            '<?php [$x, $y,] = $a;',
+        ];
+
+        yield 'group_import' => [
+            '<?php use a\{ClassA, ClassB};',
+            '<?php use a\{ClassA, ClassB,};',
+        ];
+
+        yield 'lots of nested' => [
+            '<?php $a = [1,[1,[1,[1,[1,[1,[1,[1]],[1,[1,[1,[1,[1,[1,[1,[1]]]]]]]]]]]]]];',
+            '<?php $a = [1,[1,[1,[1,[1,[1,[1,[1,],],[1,[1,[1,[1,[1,[1,[1,[1,],],],],],],],],],],],],],];',
+        ];
+    }
+
+    /**
+     * @dataProvider provideFixNoTrailingCommaInSinglelineFunctionCallCases
+     */
+    public function testFixNoTrailingCommaInSinglelineFunctionCall(string $expected, string $input = null): void
+    {
+        $this->fixer->configure(['elements' => ['arguments']]);
+
+        $this->doTest($expected, $input);
+    }
+
+    public function provideFixNoTrailingCommaInSinglelineFunctionCallCases(): iterable
+    {
+        yield 'simple var' => [
+            '<?php $a(1);',
+            '<?php $a(1,);',
+        ];
+
+        yield '&' => [
+            '<?php $a = &foo($a);',
+            '<?php $a = &foo($a,);',
+        ];
+
+        yield 'open' => [
+            '<?php foo($a);',
+            '<?php foo($a,);',
+        ];
+
+        yield '=' => [
+            '<?php $b = foo($a);',
+            '<?php $b = foo($a,);',
+        ];
+
+        yield '.' => [
+            '<?php $c = $b . foo($a);',
+            '<?php $c = $b . foo($a,);',
+        ];
+
+        yield '(' => [
+            '<?php (foo($a/* 1X */   /* 2 */  ));',
```

Comment:
IMO this should be:
```suggestion
            '<?php (foo($a /* 1X */ /* 2 */  ));',
```
i.e. removing the spaces that are directly before the comma, not directly after the previous element.

Reply:
Some fixers did one thing, others the other thing, it is a real edge case, I'm not sure it is worth the logic to handle this differently.